src/App.js

```diff
@@ -1,84 +1,86 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+
+// import weatherData from "./WeatherData";
+import Weather from "./components/Weather";
+import Loading from "./components/Loading";
+import Forecast from "./components/Forecast";
+import { filterForecast } from "./utils/helpers";
+import {cities} from "./utils/cities"
 import "./App.css";
 
 const App = () => {
+  
+  const API_KEY = process.env.REACT_APP_MY_API_ID
+  const API_BASE_CURRENT = 'https://api.openweathermap.org/data/2.5/weather?units=metric&APPID=';
+  const API_BASE_FORECAST = 'https://api.openweathermap.org/data/2.5/forecast?units=metric&APPID=';
+ 
+  // const [weather, setWeather] = useState(weatherData);
+  const [weather, setWeather] =  useState(null);
+  const [city, setCity] =  useState("Sydney");
+  const [forecast, setForecast] = useState(null);
+  
+  const fetchWeather = () => {
+    fetch(
+      API_BASE_CURRENT + API_KEY + '&q=' + city
+    )
+      .then((response) => response.json())
+      .then((json) => {
+        setWeather(json);
+        });
+  };
+
+  const fetchForecast = () => {
+    fetch (
+      API_BASE_FORECAST + API_KEY + '&q=' + city
+    )
+    .then((response) => response.json())
+    .then((json) => {
+      setForecast(filterForecast(json.list));
+    });
+  }
+
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(fetchWeather, [city]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(fetchForecast, [city]);
```

Comment:
Tohle je dobrý, ale je potřeba si uvědomit že tyhle 2 operace běží paralelně. Tj kdyby např fetchWeather trval 3s (může se někde seknout), a fetchForecast 500ms, tak se ti UI updatuje "po kouskách". Občas to chceš, ale většinou načítání dat poslouchá tenhle pattern:
- nastav loading stav
- načti z API všechna data která potřebuješ
- nastav potřebný stav a vypni loading

Plus ještě chceš ošetřit případ, kdy někdo rychle přepíná města, rychlejc než se stačí načíst. Chceš ty staré výsledky co dojdou později než nové zahodit;

tj v tomhle případě:
```
useEffect(() => {
   let cancelled = false;

   setWeather(null);
   setForecast(null);

   Promise.all([
      fetch( API_BASE_CURRENT + API_KEY + '&q=' + city).then((response) => response.json()),
fetch( API_BASE_FORECAST + API_KEY + '&q=' + city)
    .then((response) => response.json()),
   ]).then(([weather, forecast]) => {
        if(!cancelled){
            setWeather(weather);
            setForecast(filterForecast(forecast.list));
        }
   });

   return () => {
      cancelled = true;
   }
}, [city])
```
Ale to ti říkám jen pro informaci a abys viděla, že jakékoli programování s nějakými asynchroními věcmi začíná být o dost složitější :)